### PR TITLE
[TODO] Make service restart on config changes right after install

### PR DESCRIPTION
**Pull request by luitzifa**

### CODE REVIEW

**Comment by Takios** [comment]
Funktioniert so noch nicht:

```
----------
          ID: percona_server
    Function: pkg.installed
        Name: percona-server-server-5.7
      Result: True
     Comment: 1 targeted package was installed/updated.
     Started: 12:31:12.773183
    Duration: 23697.335 ms
     Changes:   
              ----------
              libaio1:
                  ----------
                  new:
                      0.3.110-2
                  old:
              libmecab2:
                  ----------
                  new:
                      0.996-1.2ubuntu1
                  old:
              libmecab2v5:
                  ----------
                  new:
                      1
                  old:
              libnuma1:
                  ----------
                  new:
                      2.0.11-1ubuntu1
                  old:
              percona-server-client-5.7:
                  ----------
                  new:
                      5.7.16-10-1.xenial
                  old:
              percona-server-common-5.7:
                  ----------
                  new:
                      5.7.16-10-1.xenial
                  old:
              percona-server-server-5.7:
                  ----------
                  new:
                      5.7.16-10-1.xenial
                  old:
----------
[...]
----------
          ID: percona_svc
    Function: service.running
        Name: mysql
      Result: True
     Comment: The service mysql is already running
     Started: 12:35:11.933590
    Duration: 112.591 ms
     Changes:   
----------
[...]
```

File: percona/config.sls [hunk 1]
```diff
@@ -42,6 +42,10 @@ mysql_python_dep:
     - watch_in:
       - service: percona_svc
 {%     endif %}
+{%     if percona_settings.reload_on_initial_conf %}
+    - require_in:
+      - service: percona_svc
+{%     endif %}
```
Comment (martin-helmich) [inline percona/config.sls:48]: To my knowledge, `watch_in` already implies `require_in`

**Comment by Takios** [comment]
The state is getting executed before writing the my.cnf:

```
----------
          ID: percona_svc
    Function: service.running
        Name: mysql
      Result: True
     Comment: The service mysql is already running
     Started: 11:44:47.782772
    Duration: 101.316 ms
     Changes:   
----------
          ID: /etc/mysql/conf.d
    Function: file.directory
      Result: True
     Comment: Directory /etc/mysql/conf.d is in the correct state
     Started: 11:44:47.887610
    Duration: 92.269 ms
     Changes:   
----------
          ID: mysql_python_dep
    Function: pkg.installed
        Name: python-mysqldb
      Result: True
     Comment: The following packages were installed/updated: python-mysqldb
     Started: 11:44:48.049692
    Duration: 10550.508 ms
     Changes:   
              ----------
              libmysqlclient20:
                  ----------
                  new:
                      5.7.16-0ubuntu0.16.04.1
                  old:
              mysql-common:
                  ----------
                  new:
                      5.7.16-0ubuntu0.16.04.1
                  old:
              mysql-common-5.6:
                  ----------
                  new:
                      1
                  old:
              python-mysqldb:
                  ----------
                  new:
                      1.3.7-1build2
                  old:
              python2.7-mysqldb:
                  ----------
                  new:
                      1
                  old:
----------
          ID: /etc/mysql/my.cnf
    Function: file.managed
      Result: True
     Comment: File /etc/mysql/my.cnf updated
     Started: 11:44:58.671717
    Duration: 984.553 ms
     Changes:   
              ----------
[...]
```

**Comment by Takios** [comment]
We're going to try a different approach.

**Pull request closed without merging**